selfservice/flow/login/handler.go

```diff
@@ -212,8 +212,30 @@ preLoginHook:
 	}
 
 	for _, s := range h.d.LoginStrategies(r.Context(), strategyFilters...) {
-		if err := s.PopulateLoginMethod(r, f.RequestedAAL, f); err != nil {
-			return nil, nil, err
+		var populateErr error
+
+		switch strategy := s.(type) {
+		case FormHydrator:
+			switch {
+			case f.IsRefresh():
+				populateErr = strategy.PopulateLoginMethodRefresh(r, f)
+			case f.RequestedAAL == identity.AuthenticatorAssuranceLevel1:
+				if h.d.Config().SelfServiceLoginFlowIdentifierFirstEnabled(r.Context()) {
+					populateErr = strategy.PopulateLoginMethodIdentifierFirstIdentification(r, f)
+				} else {
+					populateErr = strategy.PopulateLoginMethodFirstFactor(r, f)
+				}
+			case f.RequestedAAL == identity.AuthenticatorAssuranceLevel2:
+				populateErr = strategy.PopulateLoginMethodSecondFactor(r, f)
+			}
```

Comment:
```suggestion
			case f.RequestedAAL == identity.AuthenticatorAssuranceLevel2:
				populateErr = strategy.PopulateLoginMethodSecondFactor(r, f)
			case f.IsRefresh():
				populateErr = strategy.PopulateLoginMethodRefresh(r, f)
			case f.RequestedAAL == identity.AuthenticatorAssuranceLevel1:
				if h.d.Config().SelfServiceLoginFlowIdentifierFirstEnabled(r.Context()) {
					populateErr = strategy.PopulateLoginMethodIdentifierFirstIdentification(r, f)
				} else {
					populateErr = strategy.PopulateLoginMethodFirstFactor(r, f)
				}
			}
```

(the diff is messed up, essentially: move the `f.RequestedAAL == identity.AuthenticatorAssuranceLevel2:` case to be the first one, so that the MFA check has priority.)

Otherwise, a `?aal=aal2&refresh=true` flow has both MFA fields and a password input, which we have a specific test at `should go through several lookup secret lifecycles` at https://github.com/ory/kratos/blob/e0001b0db784457652581366bd7ead7cdf6b3898/test/e2e/cypress/integration/profiles/mfa/lookup.spec.ts#L128 for.

<img width="446" alt="image" src="https://github.com/ory/kratos/assets/5540255/8703aed0-45cf-47b4-af7b-fc389f884f12">

Reply:
Fix is here https://github.com/ory/kratos/pull/3960
Tested it locally and tests are passing. Let's see what CI says.

Reply:
Should be properly fixed with: https://github.com/ory/kratos/pull/3934/commits/b07329d21d1810fd3f8e5a941b72a5280f261ad0